src/aprl/envs/multi_agent.py

```diff
@@ -288,11 +288,83 @@ def _tuple_pop(input, i):
     return tuple(output), elt
 
 
+def _tuple_replace(input, i, obj):
+    output = list(input)
+    output.pop(i)
+    output.insert(i, obj)
+    return tuple(output)
+
+
 def _tuple_space_filter(tuple_space, filter_idx):
     filtered_spaces = (space for i, space in enumerate(tuple_space.spaces) if i != filter_idx)
     return gym.spaces.Tuple(tuple(filtered_spaces))
 
 
+def _tuple_space_replace(tuple_space, replace_idx, replace_space):
+    current_spaces = tuple_space.spaces
+    current_spaces.pop(replace_idx)
+    current_spaces.insert(replace_idx, replace_space)
+    return gym.spaces.Tuple(tuple(current_spaces))
+
+
+def _tuple_space_augment(tuple_space, augment_idx, augment_space):
+    current_space = tuple_space.spaces[augment_idx]
+    new_low, new_high = [np.concatenate([getattr(current_space, s),
+                                         getattr(augment_space, s)]) for s in ['low', 'high']]
+    new_space = gym.spaces.Box(low=new_low, high=new_high)
+    return _tuple_space_replace(tuple_space, augment_idx, new_space)
+
+
+class MergeAgentVecEnv(VecMultiWrapper):
+    """Allows merging of two agents into a pseudo-agent by merging their actions"""
+    def __init__(self, venv, policy, replace_action_space, merge_agent_idx):
+        """Expands one of the players in a VecMultiEnv.
+        :param env(VecMultiEnv): the environments.
+        :param action_space(): the action space of the agent to expand
+        :param agent_idx(int): the index of the agent that should be expanded into two
+        :return: a new VecMultiEnv with the same number of agents. It behaves like env but
+                 with all actions at index agent_idx set to those returned by agent."""
+        super().__init__(venv)
+
+        assert venv.num_agents >= 1  # allow currying the last agent
+        self.num_agents = venv.num_agents
```

Comment:
Oh this was a mistake, since in CurryVecEnv there's `self.num_agents = venv.num_agents - 1` and I originally just copied over that code and adjusted where necessary.